.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: ci
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        # currently windows would be difficult to support due to the build
+        # requiring autotools
+        os: [macos, ubuntu]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: "setup linux build environment"
+      if: ${{ matrix.os == 'ubuntu' }}
+      run: sudo apt install -y protobuf-compiler libprotobuf-dev libutempter-dev
+
+    - name: "setup macos build environment"
+      if: ${{ matrix.os == 'macos' }}
+      run: brew install protobuf automake
+
+    - name: "generate build scripts"
+      run: ./autogen.sh
+
+    - name: "configure"
+      run: ./configure --enable-compile-warnings=error --enable-examples
+
+    - name: "build"
```

Comment:
For both of these, we'll want the make to run in verbose mode so we can see the full invocations. I believe you can do this by running `make V=1` in mosh's build.

Reply:
Done.